linux/embedded_rootfs/pkg_addon/strongswan-4.0.4/src/_copyright/_copyright.8
.TH _COPYRIGHT 8 "25 Apr 2002"
.\"
.\" RCSID $Id: _copyright.8,v 1.1 2007-06-25 11:15:34 encounter Exp $
.\"
.SH NAME
ipsec _copyright \- prints FreeSWAN copyright
.SH DESCRIPTION
.I _copyright
outputs the FreeSWAN copyright, and version numbers for "ipsec --copyright"
.SH "SEE ALSO"
ipsec(8)
.SH HISTORY
Man page written for the Linux FreeS/WAN project
<http://www.freeswan.org/>
by Michael Richardson. Program written by Henry Spencer.
.\"
.\" $Log: not supported by cvs2svn $
.\" Revision 1.1  2004/03/15 20:35:27  as
.\" added files from freeswan-2.04-x509-1.5.3
.\"
.\" Revision 1.2  2002/04/29 22:39:31  mcr
.\" 	added basic man page for all internal commands.
.\"
.\" Revision 1.1  2002/04/26 01:21:43  mcr
.\" 	while tracking down a missing (not installed) /etc/ipsec.conf,
.\" 	MCR has decided that it is not okay for each program subdir to have
.\" 	some subset (determined with -f) of possible files.
.\" 	Each subdir that defines $PROGRAM, MUST have a PROGRAM.8 file as well as a PROGRAM file.
.\" 	Optional PROGRAM.5 files have been added to the makefiles.
.\"
.\"
.\"


linux/release-notes.txt
Release notes for Octeon SDK CN3XXX-LINUX-1.5.0

Release 1.5.0
	- Embedded root filesystem:
		- Add module-init-tools
		- Updated testsuite
		- Do a modules_prepare to get build setup for new config files
		- Add Oprofile
		- Fix path for iptables finding its shared libraries
		- Add cpu-load script for getting load percent
		- Uclibc support
		- NFS root: Fix path to ethernet driver and chroot
		- Load IPv6 kernel module if it exists
		- Fix bug where fwd rule not supported by ipsec-tools
		- Improve logic on stripping to only strip ELF files
		- Add fold for wrapping lines
		- Bug 799: Disable crypto in tcpdump with --without-crypto
		- Calculate the size of the ext3 filesystem instead of
			hardcoding a value
		- Move modules into /lib/modules/<kernel version>
		- Add iproute2 package for IPv6 support
		- Add oct-linux-profile
		- Add missing cramfs in help message
		- Add global option to turn on/off IPv6
		- Add ethtool
		- Add distclean target and make oldconfig silent
		- Fix bug where mii-tool isn't installed
		- Include user-include if it exists and a different dir
			wasn't specified. This makes it easy to add files
			without affecting the rootfs config. It is also
			more like the old minimal_rootfs.
		- Update OpenSSL version from 0.9.8b to 0.9.8d
		- Fix OpenSSL function typecast incompatibility with GCC 4.x
	- Ethernet driver:
		- Add NETPOLL support so we can support netconsole
		- Add define DONT_WRITEBACK to control if FPA frees also tell
			the L2 to throw away changes to the buffer
		- Set carrier status when link goes up or down
		- Provide help information for module parameters
		- Only enable the ethtool interface on RGMII ports
		- Fix long standing bug where RGMII ports are assigned the
			wrong PHY_ID if they're on interface 2 after an
			enabled SPI.
		- Increment RX counters for virtual ports
		- Don't use bootloader MAC addresses for virtual ports
		- Add workaround if bootloader mistakenly leaves GMX enabled
		- Set the MTU before the device comes up. It was defaulting to
			10k
		- Added parameter to skip SPI interface initialization
		- Large rewrite of the RGMII speed code. This code works for
			plug and unplug. It has trouble with unplug/plug with
			half duplex.
		- Fix crash when bridging. Don't put shared/cloned packets in
			the HW pool.
		- Remove reference to non-existent skb->tc_classid
		- Enable backpressure for SPI and RGMII
		- Set NETIF_F_LLTX - We do our own locking, Linux doesn't need
			to. This gives us a major performance boost
		- Add a new option (CVMX_ENABLE_LEN_M8_FIX) for enabling
			hardware fix for errata PKI-100
		- Cleanup simulator support. Don't do RGMII autonegotiation
			and don't read/write unmodeled CSRs.
		- Fix Bug 804. Non aligned buffers were being put in the FPA,
			causing memory corruption.
	- Intercept Module Example
		- Don't crash if ethernet driver isn't loaded. Just print a
			message and stop loading.
	- Add DRAM utilization to /proc/octeon_perf
	- Add sysn32_rt_sigtimedwait back that was removed in 2.6.16. Fixes
		N32 sigtimedwait
	- Change blocksize to 4 in NULL crypto cypher so performance isn't
		quite so bad for kernel IPSEC null encryption.
	- Add CN58XX simulator support.
	- Add CN56XX simulator support.
	- TWSI driver: Fixed problem with multi-byte read transfers (bug#793)
	- Kernel memcpy: Fix bug where prefetches were disabled.
	- Implement set_affinity for IRQs
	- PCI: Updated PCI bus bridge initialization
	- PCI: Fix PCI bug with Intel bridges causing poor performance. Intel bridges
		interpret a cache line size of 128 as an error case. Changed
		PCI cache line size to 64 to avoid this problem.			
	- PCI: Enable error detection and propagation on all PCI bridges.
	- PCI: Configure arbiter to use round robin priority instead of fixed.
		This should prevent a single device from hogging the bus.
	- PCI: Improve EBH3000 interrupt routing
	- PCI: Fix DMA mask determination in dma_map_single to correctly use
		dev->dma_mask. Fix SATA Sil 3124 driver to set DMA mask
		properly. As a side effect, the E1000 is now picked up as a
		64bit device. This fixes the problem with BAR mapping during
		unload.
	- PCI: Eliminate reads from Bar1 index registers on dma_map/unmap.
		This should be much faster.
	- PCI: Enabling smart prefetch and divided fifos for PLX 6540 bridge
	- PCI: Move BAR2 mapping code before BAR0/1 so there isn't ever an
		overlap. Clear pending PCI error interrupts after init.
	- PCI: Add workaround for PCI-302 errata
	- Debian: Refuse to destroy disks larger than 4GB
	- Debian glibc: Update to add missing .set mips0 after code Octeon added
	- Debian: Add uClibc libaries to file system.
	- Implementation of fast TLS access. See the SDK documentation for more
		information.
	- Fix Bug where a RLM interrupt on CN30XX would read Zip block, causing
		strange failures
	- Use sparsemem static instead of extreme. The extreme version uses some
		very early mem alloc that have addressing requirements. These
		can randomly corrupt low memory with any sign. The static
		version is just as good for us.
	- Add POW only ethernet driver for use with networking between cores
		running different Linux kernels. Driver is GPL.
	- CIU interrupt processing assembly was optimized to remove loops.
	- Add config options to lock often used kernel code in the L2
	- Added Octeon specific cache error exception handler.
	- Remove writing zero to L2C perf counters. They are read only.
	- Memcpy: Minor optimizations to save a few cycles
	- Use runtime checks in modules. Undefine OCTEON_MODEL so we'll get
		warnings and errors when it is used.
	- Rewrite RML interrupt handling code to handle all interrupts except
		for GXM and SPI. These should be done in the ethernet driver.
	- Add the natsemi PCI driver to the kernel config
	- Enable IPsec as modules
	- Fix oversight: enable the perfcounter interrupt. This caused Oprofile
		results to be incorrect.

linux/embedded_rootfs/pkg_addon/strongswan-4.0.4/src/_copyright/_copyright.c
/*
 * copyright reporter
 * (just avoids having the info in more than one place in the source)
 * Copyright (C) 2001  Henry Spencer.
 * 
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.  See <http://www.fsf.org/copyleft/gpl.txt>.
 * 
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * for more details.
 *
 * RCSID $Id: _copyright.c 619 2007-06-25 11:18:20Z encounter $
 */

#include <sys/types.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <getopt.h>
#include <freeswan.h>

char usage[] = "Usage: ipsec _copyright";
struct option opts[] = {
  {"help",	0,	NULL,	'h',},
  {"version",	0,	NULL,	'v',},
  {0,		0,	NULL,	0, },
};

char me[] = "ipsec _copyright";	/* for messages */

int
main(int argc, char *argv[])
{
	int opt;
	extern int optind;
	int errflg = 0;
	const char *version = ipsec_version_code();
	const char **notice = ipsec_copyright_notice();
	const char **co;

	while ((opt = getopt_long(argc, argv, "", opts, NULL)) != EOF)
		switch (opt) {
		case 'h':	/* help */
			printf("%s\n", usage);
			exit(0);
			break;
		case 'v':	/* version */
			printf("%s %s\n", me, version);
			exit(0);
			break;
		case '?':
		default:
			errflg = 1;
			break;
		}
	if (errflg || optind != argc) {
		fprintf(stderr, "%s\n", usage);
		exit(2);
	}

	for (co = notice; *co != NULL; co++)
		printf("%s\n", *co);
	exit(0);
}
